### DISCUSSION TITLE
Continuous Deploy

### DISCUSSION THREAD

**Discussion by Lory1990** [mia-platform/community#265] [Q&A] [answered]
### Which section?

Console

### Which is the version?

latest

### Description

Hi everyone,
I was wondering if there is the possibility to deploy a microservice every time I do a commit using the pipelines

The workflow cloud be this
*  Build Application
* Build Docker
* Push Docker with tag = commit
* Upload console variable
* Smart deploy

is it possibile?

**Comment by fredmaggiowski** [comment]
Hi @Lory1990 if you mean to deploy changes to your services leveraging the console you could modify your service pipelines to trigger a deploy (either from console with a service account using tge deploy api or directly within GitLab triggering the pipeline of your project configuration repository)

**Reply by peppemanzi** [reply]
You can also use miactl in your pipeline to do it: https://github.com/mia-platform/miactl/blob/main/docs/30_commands.md#deploy

**Reply by Lory1990** [reply]
My problem is that I cannot change the docker version in console.

For every deploy I would create a new image with a new tag
* myImage:12345
* myImage:8765
* myImage:2243

so i must change the image tag on console.

Moreover I think I cannot use the tag "latest" because the container would not be updated for every deploy

**Reply by peppemanzi** [reply]
You can interpolate the tag in your image url and make a commit on the `Configurations` repository to edit the `overlays/%envId%/variables.env` file or the `variables/%envId%.env` file (based on your project structure) to update the value of variable used in the interpolation. 

e.g.
Console config (sections in square brackets):
- [microservices] image url of your service: myImage:{{MY_IMAGE_VERSION}}
- [public variables] add a public variable MY_IMAGE_VERSION (keep it empty or set it to latest atm)

Service's build pipeline config:
- add deploy stage
- in deploy stage:
    - get overlays/%envId%/variables.env (or variables/%envId%.env)
    - replace MY_IMAGE_VERSION value with your new tag value
    - trigger deploy (e.g. using miactl)

**Reply by fredmaggiowski** [reply, marked as answer]
This would work, however to keep it simple you may still use latest version and force pod updates as you would normally do in the Console with the checkbox in the screenshot below.

<img width="460" alt="image" src="https://github.com/mia-platform/community/assets/7142570/c89793db-6282-4134-a17b-d595321861dd">


You can use this feature both via API or via miactl:
 - if you want to use the API you can specify the [`forceDeployWhenNoSemver`](https://console.cloud.mia-platform.eu/documentations/api-portal/#/api?search=trigger%2Fpipeline/#projects-post-%2Fapi%2Fdeploy%2Fprojects%2F%7BprojectId%7D%2Ftrigger%2Fpipeline%2F)
 - if you want to use miactl you can use the [`--no-semver`](https://github.com/mia-platform/miactl/blob/main/docs/30_commands.md#deploy)
